dcs-packaging-tool/dcs-packaging-tool-api/src/main/java/org/dataconservancy/packaging/tool/api/DomainProfileService.java

```diff
@@ -0,0 +1,163 @@
+package org.dataconservancy.packaging.tool.api;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import org.dataconservancy.packaging.tool.model.dprofile.DomainProfile;
+import org.dataconservancy.packaging.tool.model.dprofile.NodeTransform;
+import org.dataconservancy.packaging.tool.model.dprofile.PropertyType;
+import org.dataconservancy.packaging.tool.model.dprofile.PropertyValue;
+import org.dataconservancy.packaging.tool.model.ipm.Node;
+
+/**
+ * Service to perform operations on a tree of domain objects using a domain
+ * profile.
+ */
+public interface DomainProfileService {
```

Comment:
Ok I see that.

So how does that work in practice?  You instantiate a `DomainProfileServiceImpl` using a constructor that accepts an instance of `DomainProfile`?  `DomainProfile` doesn't expose any mutators.  So you'd have to instantiate it with a `DcsBoProfileImpl` for example?